### PR TITLE
Auto detect tags

### PR DESCRIPTION
**Pull request by Moejoe90**
Danger was brought back instead of codechecks. We will still use all the other codechecks features like package size etc.

### CODE REVIEW

**Comment by Moejoe90** [comment]
Will temp close

File: packages/web/package.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
     "tslint": "concurrently -p \"[{name}]\" -n \"APP,E2E\" -c \"bgGreen.bold,bgCyan.bold\" \"yarn tslint:app\" \"yarn tslint:e2e\"",
     "tslint:app": "NODE_OPTIONS=--max_old_space_size=9192 tslint --project ./tsconfig.json --format stylish",
     "tslint:e2e": "tslint --project ./tsconfig.e2e.json --format stylish",
-    "tslint:fix": "concurrently -p \"[{name}]\" -n \"APP,E2E\" -c \"bgGreen.bold,bgCyan.bold\" \"yarn tslint:app --fix\" \"yarn tslint:e2e --fix\""
+    "tslint:fix": "concurrently -p \"[{name}]\" -n \"APP,E2E\" -c \"bgGreen.bold,bgCyan.bold\" \"yarn tslint:app --fix\" \"yarn tslint:e2e --fix\"",
+    "show:tags": "echo E2E test tags && cat ./e2e-test-tags && echo"
```
Comment (arjunu) [inline packages/web/package.json]: does this need to be in package.json if it's not gonna be used outside CI?

Comment (Moejoe90) [reply packages/web/package.json]: You're question raises a good point.

File: packages/web/dangerfile.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const hasPackageLock = danger.git.created_files.includes("package-lock.json");
+
+  if (hasPackageLock) {
+    fail("Package lock detected - Do not use npm, use yarn instead.");
+  }
+  message("No Package lock detected");
+}
+
+// We will always run these tests
+const alwaysRunningTests: string = "@p1,@p2,";
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply packages/web/dangerfile.ts]: I want it to be an easy change that the QA can. 
Like this its clear what is being amended.

File: packages/web/dangerfile.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+function getE2EtestTags(): void {
+  const prBody = danger.github.pr.body;
+
+  const tags = [...new Set(prBody.replace(",", " ").match(/@\w+/g))].join(",");
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply packages/web/dangerfile.ts]: I just want the final line to not have any duplications. If the person does that in the PR its fine as long as we pass unique elements in the end. 

replacing `,` with ` ` makes it the regex simpler as you already have all the words seperated with a comma incase someone decides to split the tags with `,` 

I think its ok we don't need to error for example or something but its still good to clean it up

File: packages/web/dangerfile.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  writeFileSync("./e2e-test-tags", alwaysRunningTests + tags);
+
+  message("Tags written to to file");
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply packages/web/dangerfile.ts]: I would keep it as a general report. To make sure everything is excplicitly fine

File: packages/web/dangerfile.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+function checkPackageLock(): void {
+  if (!danger.github.pr) {
+    return;
+  }
+  const hasPackageLock = danger.git.created_files.includes("package-lock.json");
+
+  if (hasPackageLock) {
+    fail("Package lock detected - Do not use npm, use yarn instead.");
+  }
+  message("No Package lock detected");
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply packages/web/dangerfile.ts]: Why not keep explicit information?

**Pull request merged**